frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditCodeList/OptionListEditor.tsx

```diff
@@ -0,0 +1,103 @@
+  StudioErrorMessage,
+} from '@studio/components';
+import { TableIcon } from '@studio/icons';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useUpdateOptionListMutation } from '../../../../../hooks/mutations/useUpdateOptionListMutation';
+import { useOptionListsQuery } from '../../../../../hooks/queries/useOptionListsQuery';
+import { useOptionListEditorTexts } from './hooks/useOptionListEditorTexts';
+import { usePreviewContext } from 'app-development/contexts/PreviewContext';
+import classes from './OptionListEditor.module.css';
+
+type OptionListEditorProps = Pick<IGenericEditComponent<SelectionComponentType>, 'component'>;
+
+export function OptionListEditor({ component }: OptionListEditorProps): React.ReactNode {
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { data: optionsListMap, status } = useOptionListsQuery(org, app);
+
+  switch (status) {
+    case 'pending':
+      return (
+        <StudioSpinner spinnerTitle={t('ux_editor.modal_properties_code_list_spinner_title')} />
+      );
+    case 'error':
+      return (
+        <StudioErrorMessage>{t('ux_editor.modal_properties_error_message')}</StudioErrorMessage>
+      );
+    case 'success': {
+      return (
+        <OptionListEditorModal
+          optionList={optionsListMap.get(component.optionsId)}
+          component={component}
+        />
+      );
+    }
+  }
+}
+
+type OptionListEditorModalProps = {
+  optionList: Option[];
+} & Pick<IGenericEditComponent<SelectionComponentType>, 'component'>;
+
+function OptionListEditorModal({
+  optionList,
+  component,
+}: OptionListEditorModalProps): React.ReactNode {
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { doReloadPreview } = usePreviewContext();
+  const { mutate: uploadOptionList } = useUpdateOptionListMutation(org, app);
+  const [currentOptionList, setCurrentOptionList] = useState<Option[]>(optionList);
+  const editorTexts = useOptionListEditorTexts();
+  const modalRef = createRef<HTMLDialogElement>();
+
+  const handleOptionsChange = (options: Option[]) => {
+    setCurrentOptionList(options);
+  };
+
+  const handleClose = () => {
+    uploadOptionList({ optionListId: component.optionsId, optionsList: currentOptionList });
+    doReloadPreview();
+    modalRef.current?.close();
+  };
+
+  const handleClick = () => {
+    setCurrentOptionList(optionList);
+  };
+
+  return (
+    <StudioModal.Root>
+      <StudioModal.Trigger
+        onClick={handleClick}
+        className={classes.modalTrigger}
+        variant='tertiary'
+        icon={<TableIcon />}
+      >
+        {t('ux_editor.modal_properties_code_list_open_editor')}
+      </StudioModal.Trigger>
+      <StudioModal.Dialog
+        ref={modalRef}
+        className={classes.manualTabModal}
+        closeButtonTitle={t('general.close')}
+        heading={t('ux_editor.modal_add_options_codelist')}
```

Comment:
Should be `..._options_code_list` but I see there are many text keys with that error, so we can fix that later 😄